src/ACEtk/interpretation/ContinuousEnergyNeutron/Builder/EnergyDistribution/TabularLinearFunctions/Data/Builder/src/construct.hpp

```diff
@@ -0,0 +1,25 @@
+Data construct() {
+  try{
+    details::verify::equalSize( 
+      this->P_.value(), 
+      this->T_.value(),   
+      this->C_.value()
+    );
+  } catch( std::range_error& e ){
+    Log::info( "P, T, and C must all be the same size." );
+    Log::info( "P.size = {}", P_.value().size() );
+    Log::info( "T.size = {}", T_.value().size() );
+    Log::info( "C.size = {}", C_.value().size() );
+    throw;
+  }
+
+  try{
+    return { std::move( P_.value() ), 
+             std::move( T_.value() ), 
+             std::move( C_.value() ) };
+  } catch( std::bad_optional_access& b ){
+    Log::error( "Trouble creating Tabular Linear Functions data." );
+    Log::info( "Some component has not been defined." );
+    throw;
+  }
```

Comment:
Merge these two try ... catch in a single try catch. Whenever you have a bad_optional_access in the first, it will never get caught in the second.

Reply:
Good find. I've updated and improved this to be able to catch this error.